### DISCUSSION TITLE
Table breaks when there is |---|---| in first line

### DISCUSSION THREAD

**Discussion by magickaito** [markedjs/marked#2808] [Q&A] [answered]
When I have a markdown code like this:
```
Certainly! Here is a comparison of population between Singapore and Japan:
|---|---|
|Singapore |Japan |
|---|---|
|5,693,373 |126,937,352 |
```

I will get wrong output like this:

<p style="color: rgb(51, 51, 51); font-family: &quot;Helvetica Neue&quot;, Helvetica, Arial, sans-serif; font-size: medium; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;">Certainly! Here is a comparison of population between Singapore and Japan: |---|---|</p>

Singapore | Japan
-- | --
5,693,373 | 126,937,352

It looks like marked does not recognise the first line: |---|---|
[live demo](https://marked.js.org/demo/?text=Certainly!%20Here%20is%20a%20comparison%20of%20population%20between%20Singapore%20and%20Japan%3A%0A%7C---%7C---%7C%0A%7CSingapore%20%7CJapan%20%7C%0A%7C---%7C---%7C%0A%7C5%2C693%2C373%20%7C126%2C937%2C352%20%7C&options=%7B%0A%20%22async%22%3A%20false%2C%0A%20%22baseUrl%22%3A%20null%2C%0A%20%22breaks%22%3A%20false%2C%0A%20%22extensions%22%3A%20null%2C%0A%20%22gfm%22%3A%20true%2C%0A%20%22headerIds%22%3A%20true%2C%0A%20%22headerPrefix%22%3A%20%22%22%2C%0A%20%22highlight%22%3A%20null%2C%0A%20%22hooks%22%3A%20null%2C%0A%20%22langPrefix%22%3A%20%22language-%22%2C%0A%20%22mangle%22%3A%20true%2C%0A%20%22pedantic%22%3A%20false%2C%0A%20%22sanitize%22%3A%20false%2C%0A%20%22sanitizer%22%3A%20null%2C%0A%20%22silent%22%3A%20false%2C%0A%20%22smartypants%22%3A%20false%2C%0A%20%22tokenizer%22%3A%20null%2C%0A%20%22walkTokens%22%3A%20null%2C%0A%20%22xhtml%22%3A%20false%0A%7D&version=master)

Any way to solve this?

**Comment by UziTech** [comment, marked as answer]
The [spec](https://github.github.com/gfm/#tables-extension-) says the second line of a table must be the line with dashes. So that line can't be part of a table.

**Reply by UziTech** [reply]
If you want that to be part of the table you will need to create your own table tokenizer.
